src/types/reportTemplate/reportTemplate.ts

```diff
@@ -0,0 +1,364 @@
+  type: "image";
+  file_name: string;
+  url: string;
+  width: string;
+}
+
+interface RuleElement extends BaseHeaderElement {
+  type: "rule";
+  length: number;
+  stroke: string;
+}
+
+export const DateFormats = {
+  "DD/MM/YYYY": "[day]/[month]/[year]",
+  "MM/DD/YYYY": "[month]/[day]/[year]",
+  "YYYY/MM/DD": "[year]/[month]/[day]",
+  "DD-MM-YYYY": "[day]-[month]-[year]",
+  "MM-DD-YYYY": "[month]-[day]-[year]",
+  "YYYY-MM-DD": "[year]-[month]-[day]",
+  "YYYY-DD-MM": "[year]-[day]-[month]",
+  "DD/MM/YYYY hh:mm": "[day]/[month]/[year] [hour]:[minute]",
+  "MM/DD/YYYY hh:mm": "[month]/[day]/[year] [hour]:[minute]",
+  "YYYY/MM/DD hh:mm": "[year]/[month]/[day] [hour]:[minute]",
+  "DD-MM-YYYY hh:mm": "[day]-[month]-[year] [hour]:[minute]",
+  "MM-DD-YYYY hh:mm": "[month]-[day]-[year] [hour]:[minute]",
+  "YYYY-MM-DD hh:mm": "[year]-[month]-[day] [hour]:[minute]",
+  "YYYY-DD-MM hh:mm": "[year]-[day]-[month] [hour]:[minute]",
+  "DD/MM/YYYY hh:mm:ss": "[day]/[month]/[year] [hour]:[minute]:[second]",
+  "MM/DD/YYYY hh:mm:ss": "[month]/[day]/[year] [hour]:[minute]:[second]",
+  "YYYY/MM/DD hh:mm:ss": "[year]/[month]/[day] [hour]:[minute]:[second]",
+  "DD-MM-YYYY hh:mm:ss": "[day]-[month]-[year] [hour]:[minute]:[second]",
+  "MM-DD-YYYY hh:mm:ss": "[month]-[day]-[year] [hour]:[minute]:[second]",
+  "YYYY-MM-DD hh:mm:ss": "[year]-[month]-[day] [hour]:[minute]:[second]",
+  "YYYY-DD-MM hh:mm:ss": "[year]-[day]-[month] [hour]:[minute]:[second]",
+} as const;
+
+interface DateTimeElement extends BaseHeaderElement {
+  type: "datetime";
+  label: string;
+  format: string;
+  style: StyleConfig;
+}
+
+export type HeaderElementType =
+  | TextElement
+  | ImageElement
+  | RuleElement
+  | DateTimeElement;
+
+export interface HeaderRow {
+  size_ratio?: number[];
+  columns: HeaderElementType[];
+}
+
+export interface HeaderConfig {
+  rows: HeaderRow[];
+}
+
+export const FONT_WEIGHT_OPTIONS = [
+  { id: 400, value: "Normal" },
+  { id: 500, value: "Medium" },
+  { id: 600, value: "Semibold" },
+  { id: 700, value: "Bold" },
+  { id: 800, value: "Extra bold" },
+] as const;
+
+interface LabelValueField {
+  label: string;
+  value: string;
+}
+
+export interface SectionOptions {
+  title?: string;
+  fields?: string[] | Array<LabelValueField>;
+  columns?: string[];
+  style?: "list" | "text";
+  filters?: Record<string, string[]>;
+  text?: string;
+  rows?: Array<Array<string>>;
+}
+
+export interface SectionConfig {
+  source: string;
+  is_table: boolean;
+  enabled: boolean;
+  options: SectionOptions;
+}
+
+export interface ReportConfig {
+  layout: Layout;
+  header: HeaderConfig;
+  sections: SectionConfig[];
+}
+
+export const REPORT_TEMPLATE_TYPE = [
+  { id: "discharge_summary", value: "Discharge Summary" },
+  { id: "lab_report", value: "Lab Report" },
+] as const;
+
+export type ReportTemplateType = (typeof REPORT_TEMPLATE_TYPE)[number]["id"];
+
+export interface ReportTemplateBase {
+  id?: string;
```

Comment:
we are omitting id for create. so sorted for creates. but in all other usages, id will always be present right?

```suggestion
  id: string;
```